openeogeotrellis/deploy/batch_job.py

```diff
@@ -191,8 +194,65 @@ def epsg_code(gps_crs) -> Optional[int]:
                 }
             }
         else:
-            #new approach: SaveResult has generated metadata already for us
-            metadata['assets'] = asset_metadata
+            # New approach: SaveResult has generated metadata already for us
+
+            # Add the projection extension metadata.
+            # When the projection metadata is the same for all assets, then set it at the item level only.
+            # This makes it easier to read, so we see at a glance that all bands have the same proj metadata.
+            projection_metadata = {}
+
+            # We also check whether any asset file was missing or its projection metadata could not be read.
+            # In that case we never write the projection metadata at the item level.
+            missing_files_or_projection_md = False
+
+            for asset_path in asset_metadata.keys():
+                if not Path(asset_path).exists():
+                    # Path should exist so log an error.
+                    # We should only have missing files during unit tests that don't actually use files.
+                    # However, if this happens in production we should see it in the logs.
```

Comment:
Correction, I spoke to soon:
13 tests were in fact failing without the check for file existence so that means only 7 tests would be affected, not 20.
This was a test on my system, not the CI.

At first sight, some of these 13 tests rely on data that has to be mounted to the file system, and I currently don't have that on Windows Subsystem for Linux. Hence, file not found would be expected in that case.

Reply:
I have tried to check if a batch job could reach the point of extracting metadata if the asset file does not exist.

It looks like in practices the job would fail before that point, because `add_permissions(...)` would be called on the asset files, `run_job`

https://github.com/Open-EO/openeo-geopyspark-driver/blob/3d901f75a28d07ae3e03dfd127c2bcc862cf7af3/openeogeotrellis/deploy/batch_job.py#L713-L742

So, I think for now it is enough to log a warning when gdal.Info can not get metadata.

If we do add a check to verify the asset file exists, then we will have to fix a few unit tests and in future tests we would have to create some more fake files just so the thing won't raise an exception.

But if we do want that more strict checking, then let's create a follow up issue for that.